### PR TITLE
A first solution for the cli problem #4038

### LINKED ISSUE #4038: JabRef-4.2 CLI does not work
I use JabRef to sanitize my bibfile automatically. (using Jenkins ...)

the newer release 4.2 however does not work from CLI

- [x] I have tested the latest master version from http://builds.jabref.org/master/ and the problem persists

JabRef version 4.2 on Windows 10

Steps to reproduce:

using 4.1: - working
  ```
 $ java -jar JabRef-4.1.jar -n --help
usage: jabref [OPTIONS] [BIBTEX_FILE]

[...]
  ```

using 4.2: - Exception + no return to shell (same with master "4.3-dev")
  ```
$ java -jar JabRef-4.2.jar -n --help
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
        at org.jabref.cli.JabRefCLI.printUsage(JabRefCLI.java:262)
        at org.jabref.cli.ArgumentProcessor.processArguments(ArgumentProcessor.java:180)
        at org.jabref.cli.ArgumentProcessor.<init>(ArgumentProcessor.java:71)
        at org.jabref.JabRefMain.start(JabRefMain.java:112)
        at org.jabref.JabRefMain.lambda$start$1(JabRefMain.java:188)
        at java.awt.event.InvocationEvent.dispatch(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.run(Unknown Source)

  ```

### PR DESCRIPTION
**Pull request by johannes-manner**
ExportFactory is not initialized at this execution point...
The printUsage method is used twice. The JabRefCLI prints the usage, if an error occurs during argument parsing and the ArgumentProcessor prints the usage, when --help option is chosen. 

I made two solutions for this problem:
 - Commit 1: Some sort of a hack to only get the names of the exporters without altering the processing in the main method. 
image 

 - Commit 2: Changed the order of execution in the main method. The argument processor needs the export factory. I don't know, if the processing of the preferences will influence the cli functionality?

Fixes #4038 

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [X] Manually tested changed features in running JabRef

### CODE REVIEW

**Comment by stefan-kolb** [comment]
TL;DR: I guess we should use the second method. Let's see what the others think.

I guess the second solution is fine, if not, we will find out.
I think we could also extract some methods here for the initialization of the migrations, and then the preferences from the start method and afterwards instantiate the argument processor so the start method gets more clear.
